display_list/dl_builder.cc

```diff
@@ -1786,6 +1812,63 @@ bool DisplayListBuilder::AccumulateBounds(const SkRect& bounds,
   return true;
 }
 
+bool DisplayListBuilder::LayerInfo::update_clear_color(DlColor color,
```

Comment:
The render target's clear color will ignore all clips, so any time we add a non-culled clip to the builder we have to turn off clear color optimization.

This is makes clip culling is super important

Reply:
Because of this actually, we may need to do clear color optimization while dispatching instead of recording, but lets discuss today and see if folks have ideas.

Reply:
I thought of that last night. I was thinking that turning off the optimization after receiving a clip would do the trick there, but there is also the fact that the DL Builder is still figuring out the right bounds for the save layer while it is recording and won't know the true bounds until Restore is called. So, perhaps this entire optimization is not possible. Or perhaps it only works for DrawColor and DrawPaint that happen right after a SaveLayer but before any clips.

We'll discuss later...

Reply:
I should probably transfer all of the new code to the clip functions to cull unnecessary clips even if it doesn't get used for clear color overlap detection...